android/Staccato_AN/app/src/main/java/com/woowacourse/staccato/presentation/memorycreation/MemoryCreationActivity.kt

```diff
@@ -105,6 +116,23 @@ class MemoryCreationActivity : BindingActivity<ActivityMemoryCreationBinding>(),
         }
     }
 
+    private fun setHidingKeyboardAction() {
+        binding.root.setOnClickListener {
+            hideKeyboard()
+        }
+    }
+
+    private fun hideKeyboard() {
+        if (this.currentFocus != null) {
+            val inputManager: InputMethodManager =
+                this.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
+            inputManager.hideSoftInputFromWindow(
+                this.currentFocus?.windowToken,
+                InputMethodManager.HIDE_NOT_ALWAYS,
+            )
+        }
+    }
```

Comment:
```suggestion
        private fun hideKeyboard() {
        if (currentFocus != null) {
            inputManager.hideSoftInputFromWindow(
                currentFocus?.windowToken,
                InputMethodManager.HIDE_NOT_ALWAYS,
            )
        }
    }
```
자바와 달리 kotlin은 `this`를 생략할 수 있다는 점이 큰 장점이라고 생각해요. `this`는 생략하는 게 좋을 것 같아용!

그리고 `InputMethodManager`의 초기화는 아래와 같이 `by lazy`를 활용하는 방법을 사용해도 괜찮을 것 같은데, 호두는 이에 대해 어떻게 생각하시는 지 궁금합니다!

```kotlin
    private val inputManager: InputMethodManager by lazy {
        getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
    }
```

Reply:
너무 좋네요...!! 처음 참고했던 코드가 자바 코드여서, 그대로 긁어오다보니 자바의 흔적이 여럿 남아있네요...

추가로 `InputMethodManager`에 대해 알아보니, 하나의 Application 마다 존재하며, 싱글톤처럼 어플리케이션 전역에서 위치하여 사용된다고 합니다. 따라서 호출 시 기존에 어플리케이션에 존재하는 인스턴스를 가져오게 됩니다.

하지만 이전 코드에서는 키보드 바깥의 터치가 이루어질 때 마다 매번 `InputMethodManager` 인스턴스를 가져와 변수로서 저장하는 동작을 하므로, 불필요한 변수 설정이 이루어지는 것 같습니다.

해나가 제시한 지연 초기화 방식이 코드 가독성과 동작의 측면에서 이점이 있겠네요! 적용하여 리팩터링 해보겠습니다 👍